contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,268 @@
+        uint8 opponentMove;
+        uint256 expiryDate;
+    }
+
+    event Deposit(
+        address player,
+        uint256 amount
+    );
+
+    event GameCreated(
+        address indexed creator,
+        bytes32 indexed gameToken,
+        uint256 stake
+    );
+
+    event OpponentPlays(
+        address indexed opponent,
+        bytes32 indexed token,
+        uint8 move
+    );
+
+    event Draw(
+        bytes32 indexed token
+    );
+
+    event WinnerRewarded(
+        bytes32 indexed gameToken,
+        address indexed winner,
+        address indexed loser,
+        uint256 winnings,
+        uint8 creatorMove,
+        uint8 opponentMove
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint amount
+    );
+
+    event  CreatorEndsGame(
+        address indexed player,
+        bytes32 indexed gameToken,
+        uint256 refund
+    );
+
+    event  ForfeitPaid(
+        address indexed player,
+        bytes32 indexed gameToken,
+        uint256 forfeit
+    );
+
+    function deposit() external payable whenNotPaused {
+
+        balances[msg.sender] = balances[msg.sender].add(msg.value);
+
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    function create(bytes32 playerMoveHash, uint stake) external payable whenNotPaused  returns(bytes32){
+
+        require(playerMoveHash != bytes32(0), "Player move cannot be empty");
+        require(stake > 0, "Stake cannot be zero");
+
+        bytes32 gameToken = keccak256(abi.encodePacked(msg.sender, block.timestamp));
+
+        Game storage game = games[gameToken];
+        require(game.creator == NULL_ADDRESS, "Token already in use");
+
+        game.creator = msg.sender;
+        game.stake = stake;
+        game.moveHash = playerMoveHash;
+        fundMove(stake);
+        emit GameCreated(msg.sender, gameToken, stake);
+
+        return gameToken;
+    }
+
+    /*
+    * The opponent makes their move and waits for the creator to resolve the game by revealing their move
+    */
+    function opponentPlay(bytes32 gameToken, uint8 playerMove) external payable whenNotPaused {
+
+        Game storage game = games[gameToken];
+        uint256 stake = game.stake;
+        require(stake != 0, GAME_NOT_FOUND_MSG);
+        require(game.opponent == NULL_ADDRESS, "Opponent has already moved");
+        require(playerMove < 3, INVALID_MOVE_MSG);
+
+        game.opponentMove = playerMove;
+        game.expiryDate =  block.timestamp + FORFEIT_WINDOW;
+        fundMove(stake);
+        emit OpponentPlays(msg.sender, gameToken, playerMove);
+    }
+
+
+    function fundMove(uint256 stake) internal {
+        uint256 senderBalance = balances[msg.sender];
+
+        if (msg.value < stake) {
+            require(balances[msg.sender] >= stake - msg.value, "Insufficient balance");
+            balances[msg.sender] = SafeMath.sub(senderBalance, stake - msg.value);
+            return;
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value - stake);
```

Comment:
Instead of a mid-flow return, you could have an `if else`, that would be clearer I believe.
Otherwise, since you did not do any optimisation for `msg.value == stake`, do you see how the same could (almost) be done in 1 line?

Reply:
Yes. I think so.